cmd/Makefile.am

```diff
@@ -128,14 +132,23 @@ noinst_PROGRAMS += decode-mount-opts/decode-mount-opts
 decode_mount_opts_decode_mount_opts_SOURCES = \
 	decode-mount-opts/decode-mount-opts.c
 decode_mount_opts_decode_mount_opts_LDADD = libsnap-confine-private.a
+decode_mount_opts_decode_mount_opts_STATIC =
+
+if STATIC_LIBCAP
+decode_mount_opts_decode_mount_opts_STATIC += -lcap
+else
+decode_mount_opts_decode_mount_opts_LDADD += -lcap
```

Comment:
@niemeyer not quite, if you want to link to static libraries via `.a` files you have to spell out their full location.  As a quick hack I did this:
```
diff --git a/cmd/Makefile.am b/cmd/Makefile.am
index 695b07e..982543c 100644
--- a/cmd/Makefile.am
+++ b/cmd/Makefile.am
@@ -110,17 +110,10 @@ libsnap_confine_private_unit_tests_CFLAGS += -D_ENABLE_FAULT_INJECTION
 libsnap_confine_private_unit_tests_STATIC =
 
 if STATIC_LIBCAP
-libsnap_confine_private_unit_tests_STATIC += -lcap
+libsnap_confine_private_unit_tests_LDADD += /usr/lib/x86_64-linux-gnu/libcap.a
 else
 libsnap_confine_private_unit_tests_LDADD += -lcap
 endif  # STATIC_LIBCAP
-
-# Use a hacked rule if we're doing static build. This allows us to inject the LIBS += .. rule below.
-libsnap-confine-private/unit-tests$(EXEEXT): $(libsnap_confine_private_unit_tests_OBJECTS) $(libsnap_confine_private_unit_tests_DEPENDENCIES) $(EXTRA_libsnap_confine_private_unit_tests_DEPENDENCIES) libsnap-confine-private/$(am__dirstamp)
-       @rm -f libsnap-confine-private/unit-tests$(EXEEXT)
-       $(AM_V_CCLD)$(libsnap_confine_private_unit_tests_LINK) $(libsnap_confine_private_unit_tests_OBJECTS) $(libsnap_confine_private_unit_tests_LDADD) $(LIBS)
-
-libsnap-confine-private/unit-tests$(EXEEXT): LIBS += -Wl,-Bstatic $(libsnap_confine_private_unit_tests_STATIC) -Wl,-Bdynamic
 endif  # WITH_UNIT_TESTS
 
 ##
```

The results are correct but the onus of finding the library is on us. If we use `-lcap` wrapped with linker options to link statically (like we currently do) it just works.
```
zyga@fyke:~/snapd/cmd$ ldd libsnap-confine-private/unit-tests
	linux-vdso.so.1 =>  (0x00007fff18bfd000)
	libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f414a0a3000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f4149cda000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f4149a69000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f414984c000)
	/lib64/ld-linux-x86-64.so.2 (0x0000562020b78000)
```